### PR TITLE
.Net: Claude tools models

### PR DESCRIPTION
**Pull request by Krzysztof318**
Claude tools models

cc: @RogerBarreto 

### Contribution Checklist

- [x] The code builds clean without any errors or warnings
- [x] The PR follows the [SK Contribution Guidelines](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md) and the [pre-submission formatting script](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md#development-scripts) raises no violations
- [x] All unit tests pass, and I have added new tests where possible
- [x] I didn't break anyone :smile:

### CODE REVIEW

**Comment by RogerBarreto** [comment]
Anthropic has one API for its models, use the `Anthropic` Prefix instead of Claude for classes.

### `Anthropic` API 
image

### Model can vary (Claude, etc)
image

Observation, their versioning works in the header.

**Comment by Krzysztof318** [comment]
@RogerBarreto I didn't know they have common api for all models, thanks.

**Comment by Krzysztof318** [comment]
I renamed all except tool/function related classes. What do you think, should also rename function calling classes?

File: dotnet/src/Connectors/Connectors.Anthropic/Core/Models/Message/AnthropicImageContent.cs [hunk 1]
```diff
@@ -4,31 +4,19 @@
 
 namespace Microsoft.SemanticKernel.Connectors.Anthropic.Core;
 
-/// <summary>
-/// Represents the request/response content of Claude.
-/// </summary>
-internal sealed class ClaudeMessageContent
+internal sealed class AnthropicImageContent : AnthropicMessageContent
```
Comment (RogerBarreto) [inline dotnet/src/Connectors/Connectors.Anthropic/Core/Models/Message/AnthropicImageContent.cs:7]: As this is not a KernelImageContent, suggest using something more related to another concept as `ImageMessage` as this is also a `Message`.

You sure the Image has all parameters of a message?

```suggestion
internal sealed class AnthropicImageMessageContent : AnthropicMessageContent
```

File: dotnet/src/Connectors/Connectors.Anthropic/Core/Models/Message/AnthropicMessageContent.cs [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+/// <summary>
+/// Represents the request/response content of Claude.
+/// </summary>
+[JsonPolymorphic(TypeDiscriminatorPropertyName = "type")]
+[JsonDerivedType(typeof(AnthropicTextContent), typeDiscriminator: "text")]
+[JsonDerivedType(typeof(AnthropicImageContent), typeDiscriminator: "image")]
+[JsonDerivedType(typeof(AnthropicToolCallContent), typeDiscriminator: "tool_use")]
+[JsonDerivedType(typeof(AnthropicToolResultContent), typeDiscriminator: "tool_result")]
+internal abstract class AnthropicMessageContent { }
```
Comment (RogerBarreto) [inline dotnet/src/Connectors/Connectors.Anthropic/Core/Models/Message/AnthropicMessageContent.cs:15]:
```suggestion
internal abstract class AnthropicContent { }
```

File: dotnet/src/Connectors/Connectors.Anthropic/Core/Models/Message/AnthropicTextContent.cs [hunk 3]
```diff
@@ -0,0 +1,20 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System.Text.Json.Serialization;
+
+namespace Microsoft.SemanticKernel.Connectors.Anthropic.Core;
+
+internal sealed class AnthropicTextContent : AnthropicMessageContent
```
Comment (RogerBarreto) [inline dotnet/src/Connectors/Connectors.Anthropic/Core/Models/Message/AnthropicTextContent.cs:7]:
```suggestion
internal sealed class AnthropicTextContent : AnthropicContent
```

File: dotnet/src/Connectors/Connectors.Anthropic.UnitTests/Utils/ClaudeKernelFunctionMetadataExtensions.cs [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+using System.Collections.Generic;
+using Microsoft.SemanticKernel;
+using Microsoft.SemanticKernel.Connectors.Anthropic;
+
+namespace SemanticKernel.Connectors.Anthropic.UnitTests;
+
+/// <summary>
+/// Extensions for <see cref="KernelFunctionMetadata"/> specific to the Claude connector.
+/// </summary>
+public static class ClaudeKernelFunctionMetadataExtensions
```
Comment (RogerBarreto) [inline dotnet/src/Connectors/Connectors.Anthropic.UnitTests/Utils/ClaudeKernelFunctionMetadataExtensions.cs:12]:
```suggestion
public static class AnthropicKernelFunctionMetadataExtensions
```

**Pull request merged**